tests/main/interfaces-desktop-host-fonts/task.yaml

```diff
@@ -0,0 +1,45 @@
+summary: Ensure that the desktop interface gives access to host fonts
+
+details: |
+    In order to ensure that confined applications have access to fonts
+    covering the user's spoken language, the host system's fonts are
+    bind mounted into the sandbox.
+
+systems: [-ubuntu-core-*]
+
+execute: |
+    mkdir -p /usr/share/fonts
+    echo "Distribution font" > /usr/share/fonts/dist-font.txt
+
+    mkdir -p /usr/local/share/fonts
+    echo "Local font" > /usr/local/share/fonts/local-font.txt
+
+    mkdir -p /var/cache/fontconfig
+    echo "Cache file" > /var/cache/fontconfig/cache.txt
+
+    mkdir -p $HOME/.fonts
+    echo "User font 1" > $HOME/.fonts/user-font1.txt
+
+    mkdir -p $HOME/.local/share/fonts
+    echo "User font 2" > $HOME/.local/share/fonts/user-font2.txt
+
+    echo "Install the test-snapd-desktop snap"
+    snap try $TESTSLIB/snaps/test-snapd-desktop
+
+    echo "The plug is connected by default"
+    snap interfaces | MATCH ":desktop .*test-snapd-desktop"
+
+    echo "Checking access to host /usr/share/fonts"
+    test-snapd-desktop.check-files /usr/share/fonts/dist-font.txt | MATCH "Distribution font"
+
+    echo "Checking access to host /usr/local/share/fonts"
+    test-snapd-desktop.check-files /usr/local/share/fonts/local-font.txt | MATCH "Local font"
+
+    echo "Checking access to host /var/cache/fontconfig"
+    test-snapd-desktop.check-files /var/cache/fontconfig/cache.txt | MATCH "Cache file"
+
+    echo "Checking access to host ~/.fonts"
+    test-snapd-desktop.check-files $HOME/.fonts/user-font1.txt | MATCH "User font 1"
```

Comment:
HOME will be expanded by the shell running spread. This will not really work "in practice" `check-file` were to expand HOME (since HOME is changed by snap-run chain). I wonder if we should put a symlink in the remapped `$HOME/.fonts` (that is `$SNAP_USER_DATA`) to vanilla `$HOME/.fonts`.

Reply:
That's the intention here: we want to check that the sandboxed app can access the real ~/.fonts and ~/.local/share/fonts to see any fonts the user has personally installed.  We don't want $SNAP_USER_DATA here.

Reply:
Yeah but what will make snaps use the real HOME/.fonts?

Reply:
For snaps using the various desktop-launch script from snapcraft-desktop-helpers, we configure fontconfig to look for fonts in the real home directory:

https://github.com/ubuntu/snapcraft-desktop-helpers/blob/master/common/desktop-exports#L214-L235